### PR TITLE
[READY FOR REVIEW] Add JellySwap Spells  

### PR DESCRIPTION
**Pull request by viniabussafi**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

---

JellySwap is a friendly fork of Balancer V2 on Sei. This PR recreates the whole balancer_v2 spell lineage, including the trades spell, with no changes introduced to the spells themselves.
Also had to create some dex.trades dependencies for sei

### CODE REVIEW

File: dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql
Comment (Hosuke) [inline dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql:14]: Do you think we may simplify this macro?

**Review by Hosuke — APPROVED** [review]
Thank you @viniabussafi

**Comment by jeff-dude** [comment]
wow, fantastic! @Hosuke was actually going to try to build in #6423 but we're more than happy to have the experts help us build instead 🙏

File: dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql
Comment (viniabussafi) [reply dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql:14]: only low hanging fruit I see is discarding this parameter:
`Vault_evt_Swap = 'Vault_evt_Swap'`
Open to any other suggestions you may have, though

Comment (jeff-dude) [reply dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql:14]: lets save any potential refactor of a macro to separate PR. i'm also comfortable with vini having final say on the balancer macros, as our lead on that 🙏

File: dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql
Comment (jeff-dude) [inline dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql:39]:
```suggestion
    {% else %}
```

File: dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql
Comment (jeff-dude) [inline dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql:40]:
```suggestion
    AND {{ incremental_predicate('logs.block_time') }}
```

File: dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql
Comment (jeff-dude) [inline dex/models/_projects/jelly_swap/jelly_swap_sei_pools_fees.sql:8]:
```suggestion
        unique_key = ['block_number', 'tx_hash', 'index'],
        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
```

File: dex/models/_projects/jelly_swap/jelly_swap_sei_trades.sql
Comment (jeff-dude) [inline dex/models/_projects/jelly_swap/jelly_swap_sei_trades.sql:1]: since this isn't chain specific (in case they go crosschain), maybe file rename to remove `sei`?

File: dex/models/_projects/jelly_swap/jelly_swap_sei_trades.sql
Comment (jeff-dude) [inline dex/models/_projects/jelly_swap/jelly_swap_sei_trades.sql:8]:
```suggestion
        materialized = 'view'
```

Comment (jeff-dude) [reply dex/models/_projects/jelly_swap/jelly_swap_sei_trades.sql:8]: we're leaning towards displaying less in the data explorer up front. open to adding, if there is strong case, but most users will query `dex.trades` and filter on projects rather than use these views directly

File: dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql
Comment (jeff-dude) [inline dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql:3]:
```suggestion
        schema = 'jelly_swap_sei',
```

Comment (jeff-dude) [reply dex/models/trades/sei/platforms/jelly_swap_sei_base_trades.sql:3]: should we remove `v2` to match file name? or do the opposite and keep version but add to files?

File: models/_sector/labels/addresses/__single_category_labels__/jelly_swap/labels_jelly_swap_pools_sei.sql
Comment (jeff-dude) [inline models/_sector/labels/addresses/__single_category_labels__/jelly_swap/labels_jelly_swap_pools_sei.sql:7]:
```suggestion
  alias = 'jelly_swap_pools_sei'
```

File: models/jelly_swap/sei/jelly_swap_sei_liquidity.sql
Comment (jeff-dude) [inline models/jelly_swap/sei/jelly_swap_sei_liquidity.sql:5]:
```suggestion
        schema = 'jelly_swap_' + blockchain,
```

Comment (jeff-dude) [reply models/jelly_swap/sei/jelly_swap_sei_liquidity.sql:5]: not a must, just an fyi on how you can use 👍

File: models/jelly_swap/sei/jelly_swap_sei_pools_tokens_weights.sql
Comment (jeff-dude) [inline models/jelly_swap/sei/jelly_swap_sei_pools_tokens_weights.sql:21]:
```suggestion
    WHERE {{ incremental_predicate('evt_block_time') }}
```

File: models/jelly_swap/sei/jelly_swap_sei_pools_tokens_weights.sql
Comment (jeff-dude) [inline models/jelly_swap/sei/jelly_swap_sei_pools_tokens_weights.sql:35]:
```suggestion
    AND {{ incremental_predicate('call_create.call_block_time') }}
```

File: sources/_subprojects/dex/_sources.yml
Comment (jeff-dude) [inline sources/_subprojects/dex/_sources.yml:73]: i know this is likely confusing, and we will look to document along with how to use subprojects in general, but we should list spells that are sources cross-project to be placed in directory path of where the spell is built. we will be moving `labels` spells to `daily_spellbook`, so you could proactively add to that file?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by viniabussafi** [comment]
thanks for the review, @jeff-dude! should be better now

**Comment by jeff-dude** [comment]
it's possible this latest failure is due to a broken price feed we just fixed. trying CI again.

**Review by jeff-dude — APPROVED** [review]
approving pending CI finish 🤝

**Comment by jeff-dude** [comment]
> approving pending CI finish 🤝

not sure why CI is so slow on this one, but we got past the important parts. will try the merge.

**Pull request merged**
